### PR TITLE
[SPARK-41176][SQL] Assign a name to the error class _LEGACY_ERROR_TEMP_1042

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
In the PR, I propose to assign a name to the error class _LEGACY_ERROR_TEMP_1042.

### Why are the changes needed?
Proper names of error classes should improve user experience with Spark SQL.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala [hunk 1]
```diff
@@ -637,13 +637,13 @@ private[sql] object QueryCompilationErrors extends QueryErrorsBase {
...
-      name: String, expectedInfo: String, actualNumber: Int): Throwable = {
+      name: String, expectedNum: String, actualNum: Int): Throwable = {
     new AnalysisException(
-      errorClass = "_LEGACY_ERROR_TEMP_1042",
+      errorClass = "FUNCTION_WRONG_NUM_ARGS",
       messageParameters = Map(
-        "name" -> name,
-        "expectedInfo" -> expectedInfo,
-        "actualNumber" -> actualNumber.toString))
+        "functionName" -> name,
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryCompilationErrors.scala:644]: Please, wrap `functionName` by `toSQLId()`.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -490,6 +490,11 @@
       "The operation <statement> is not allowed on the <objectType>: <objectName>"
     ]
   },
+  "FUNCTION_WRONG_NUM_ARGS" : {
```
[the remark this answers is not in the document]
Comment (srielau) [reply core/src/main/resources/error/error-classes.json:513]: WRONG_NUM_ARGs is fine. If we want to keep a longer name I'd prefer ROUTINE over FUNCTIOn for future proofing.

[the remark this answers is not in the document]
Comment (MaxGekk) [reply core/src/main/resources/error/error-classes.json:513]: @panbingkun Please, rename it to `WRONG_NUM_ARGS`.

[the remark this answers is not in the document]
Comment (panbingkun) [reply core/src/main/resources/error/error-classes.json]: Done

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @panbingkun and @srielau for review.

**Pull request closed without merging**